### PR TITLE
Enable connecting to the AWS IoT MQTT broker

### PR DESCRIPTION
**Pull request by sbeaulie**
See README file  for changelog

See amazon doc for details of what client configuration they support
http://docs.aws.amazon.com/iot/latest/developerguide//protocols.html

### CODE REVIEW

File: src/main/java/org/apache/jmeter/protocol/mqtt/paho/clients/BlockingClient.java [hunk 1]
```diff
@@ -64,24 +64,21 @@ public BlockingClient(String brokerUrl, String clientId, boolean cleanSession, S
...
-        String testPlanFileDir = System.getProperty("java.io.tmpdir");
-
-        if (null != GuiPackage.getInstance()) {
-            String testPlanFile = GuiPackage.getInstance().getTestPlanFile();
-            testPlanFileDir = FilenameUtils.getFullPathNoEndSeparator(testPlanFile);
-            testPlanFileDir = testPlanFileDir + File.separator + "tmp" + File.separator + clientId + File.separator + Thread.currentThread().getId();
-        }
+        //String tmpDir = System.getProperty("java.io.tmpdir");
 
+        String testPlanFile = GuiPackage.getInstance().getTestPlanFile();
```
Comment (hemikak) [inline src/main/java/org/apache/jmeter/protocol/mqtt/paho/clients/BlockingClient.java:69]: If we are running headless, "GuiPackage.getInstance()" will give a nullpointerexception.

File: src/main/java/org/apache/jmeter/protocol/mqtt/sampler/PublisherSampler.java [hunk 2]
```diff
@@ -160,6 +162,7 @@ public void threadStarted() {
         parameters.addArgument("MESSAGE_VALUE", getMessageValue());
 
         this.context = new JavaSamplerContext(parameters);
+        this.publisher.setNameLabel(getName());
```
Comment (hemikak) [inline src/main/java/org/apache/jmeter/protocol/mqtt/sampler/PublisherSampler.java:165]: +1 for this. I missed it out.

**Comment by hemikak** [comment]
Can you get the latest of the README.md file and commit. I made some changes it recently.

File: pom.xml [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 	<properties>
 		<jmeter-version>2.10</jmeter-version>
-		<eclipse.paho.mqtt.version>0.4.0</eclipse.paho.mqtt.version>
+		<eclipse.paho.mqtt.version>1.0.2</eclipse.paho.mqtt.version>
```
Comment (hemikak) [inline pom.xml:16]: +1

File: src/main/java/org/apache/jmeter/protocol/mqtt/paho/clients/BlockingClient.java (hunk 1 above)
Comment (sbeaulie) [inline src/main/java/org/apache/jmeter/protocol/mqtt/paho/clients/BlockingClient.java:69]: Thanks that will be a problem for me too.  We can use
FileServer.getFileServer().getScriptName(); instead

**Comment by sbeaulie** [comment]
If you merge you can keep your version of the file it was just for me to
write down the changes made to your script. I see you have made changes in
parallel to the master version I used so feel free to keep your changes too

**Pull request merged**